File: day01/kotlin/Main.kt
import java.io.File

/**
 * Advent of Code 2025 - Day 01
 * 
 * How to run:
 * 1. Using kotlinc (compile and run):
 *    kotlinc Main.kt -include-runtime -d Main.jar
 *    java -jar Main.jar
 * 
 * 2. Using kotlin script:
 *    kotlin Main.kt
 * 
 * 3. Using IntelliJ IDEA or other IDE:
 *    Open this file and run the main function
 */

fun main() {
    // Read input file (assumes input.txt is in the parent directory)
    val inputFile = File("../input.txt")
    
    if (!inputFile.exists()) {
        println("Error: input.txt not found")
        return
    }
    
    val lines = inputFile.readLines()
    
    println("Day 01 Solution")
    println("===============")
    println("Read ${lines.size} lines from input.txt")
    println()
    
    // Process each line
    lines.forEach { line ->
        // TODO: Implement your solution here
        println("Line: $line")
    }
    
    // Part 1
    val part1Result = solvePart1(lines)
    println("Part 1: $part1Result")
    
    // Part 2
    val part2Result = solvePart2(lines)
    println("Part 2: $part2Result")
}

fun solvePart1(lines: List<String>): Any {
    // TODO: Implement Part 1 solution
    return "Not implemented yet"
}

fun solvePart2(lines: List<String>): Any {
    // TODO: Implement Part 2 solution
    return "Not implemented yet"
}

File: day01/haskell/Main.hs
{-
  Advent of Code 2025 - Day 01

  How to run:
  cabal run
-}

module Main where

import Control.Exception (catch, IOException)

main :: IO ()
main = do
  putStrLn "Day 01 Solution"
  putStrLn "==============="

  -- Read input file (assumes input.txt is in the parent directory)
  contents <- readInputFile "../input.txt"

  case contents of
    Nothing -> putStrLn "Error: Could not read input.txt"
    Just input -> do
      let inputLines = lines input
      putStrLn $ "Read " ++ show (length inputLines) ++ " lines from input.txt"
      putStrLn ""

      -- Part 1
      let part1Result = solvePart1 inputLines
      putStrLn $ "Part 1: " ++ show part1Result

      -- Part 2
      let part2Result = solvePart2 inputLines
      putStrLn $ "Part 2: " ++ show part2Result

-- Helper function to read input file with error handling
readInputFile :: FilePath -> IO (Maybe String)
readInputFile path =
  catch (Just <$> readFile path)
        (\e -> const (return Nothing) (e :: IOException))

parseLine :: String -> (Char, Int)
parseLine (d:v) = (d, read v)
parseLine _ = error "Invalid input line"

-- Part 1 solution
solvePart1 :: [String] -> Int
solvePart1 inputLines =
  let
    moves = map parseLine inputLines
    (_, hits) = foldl step (50, 0) moves
    step :: (Int, Int) -> (Char, Int) -> (Int, Int)
    step (pos, cumulatedHits) (dir, val) =
      let newPos = case dir of
            'R' -> pos + val
            'L' -> pos - val
            _   -> error $ "Unknown direction: " ++ [dir]
          newHits = if newPos `mod` 100 == 0 then cumulatedHits + 1 else cumulatedHits
      in (newPos, newHits)
  in hits

-- Part 2 solution
solvePart2 :: [String] -> String
solvePart2 _inputLines =
  -- TODO: Implement Part 2 solution
  "Not implemented yet"
